apps/web/screens/LifeEvents/LifeEvents.tsx

```diff
@@ -0,0 +1,154 @@
+import Head from 'next/head'
+import {
+  Text,
+  GridContainer,
+  GridRow,
+  GridColumn,
+  Breadcrumbs,
+} from '@island.is/island-ui/core'
+import {
+  LinkType,
+  linkResolver,
+  useNamespaceStrict as useNamespace,
+} from '@island.is/web/hooks'
+import { withMainLayout } from '@island.is/web/layouts/main'
+import {
+  type QueryGetNamespaceArgs,
+  type GetNamespaceQuery,
+  GetLifeEventsQuery,
+} from '@island.is/web/graphql/schema'
+import type { Screen } from '../../types'
+import { CardWithFeaturedItems, GridItems } from '@island.is/web/components'
+import { ContentLanguage, QueryGetLifeEventsArgs } from '@island.is/api/schema'
+import { GET_LIFE_EVENTS_QUERY, GET_NAMESPACE_QUERY } from '../queries'
+import { ApplicationsTexts } from './LifeEvents.types'
+
+type LifeEvents = GetLifeEventsQuery['getLifeEvents']
```

Comment:
is this the proper file for assigning this type?

Reply:
@busla Can you explain what you mean? The ApplicationsText type is for using the strict namespace for i18n. This is following how it's been set up in other parts of the web app. The `GetLifeEventsQuery['getLifeEvents']` type is simply inferred from the graphql call.

Reply:
@busla After some quick investigations, it looks like, for whatever reason, the codegen for this specific query is not working and thus, the type that is being referenced is an old type for this query. Prior to codegen, the generated query looks like this:

```
export type GetLifeEventsQuery = {
  __typename?: 'Query'
  getLifeEvents: Array<{
    __typename?: 'LifeEventPage'
    id: string
    title: string
    slug: string
    intro?: string | null
    thumbnail?: { __typename?: 'Image'; url: string; title: string } | null
    image?: { __typename?: 'Image'; url: string; title: string } | null
  }>
}
```

This matches exactly the type check that is failing during the build process. I can't yet see why this is happening but after local codegen, the type correctly is set to

```
export type GetLifeEventsQuery = {
  __typename?: 'Query'
  getLifeEvents: Array<{
    __typename?: 'LifeEventPage'
    id: string
    title: string
    shortTitle?: string | null
    slug: string
    seeMoreText?: string | null
    tinyThumbnail?: { __typename?: 'Image'; url: string; title: string } | null
    featured: Array<{
      __typename?: 'Featured'
      title: string
      attention: boolean
      thing?: {
        __typename?: 'ReferenceLink'
        slug: string
        type: string
      } | null
    }>
  }>
}
```

Reply:
We only run `codegen` in CI when files change according to the [these glob patterns](https://github.com/island-is/island.is/blob/main/scripts/_hash-generated-files.sh).

This file is not included so a generated files cache is downloaded, missing the attribute.

We can include this file in the list but I wonder if the type should be defined within a file that we are already checking for.

Reply:
i think it did the trick @busla

Reply:
I'm still curious why it wasn't building the other way. I don't see a specific reason for it and am wondering if it was a caching issue with nx perhaps? The only thing that has changed is the naming of the query and resolver/service methods. Oh well, there's a working build. We will remove the old, unused methods and query.